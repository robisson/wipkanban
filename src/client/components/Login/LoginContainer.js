import { connect } from "react-redux";
import { login } from "../../actions/Login";
import FormLogin from "./index";

const mapDispatchToProps = dispatch => {
  return {
    onLogin(email, password) {
      dispatch(login(email.value, password.value));
    }
  };
};

const mapStateToProps = state => ({ state: state.userReducer });

const FormLoginContainer = connect(mapStateToProps, mapDispatchToProps)(
  FormLogin
);

export default FormLoginContainer;
