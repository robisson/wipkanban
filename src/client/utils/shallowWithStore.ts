import { shallow } from "enzyme";

export default (component:any, store:any) => {
  const context = {
    store
  };
  return shallow(component, { context });
};
