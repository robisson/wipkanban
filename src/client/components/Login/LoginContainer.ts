import { connect } from "react-redux";
import { login, oauthFacebook } from "../../actions/Login";
import FormLogin from "./index";
import { Dispatch } from "redux";

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    onLogin(email: string, password: string) {
      dispatch(login(email, password));
    },
    oauthFacebook(accessToken: string) {
      dispatch(oauthFacebook(accessToken));
    }
  };
};

const mapStateToProps = (state:any) => ({ state: state.userReducer });

const FormLoginContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(FormLogin);

export default FormLoginContainer;
