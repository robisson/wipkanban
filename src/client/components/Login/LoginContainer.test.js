import React from "react";
import toJson from "enzyme-to-json";
import "../../setupTest";
import LoginContainer from "./LoginContainer";
import shallowWithStore from "../../utils/shallowWithStore";
import { createMockStore } from "redux-test-utils";

describe("<LoginContainer /> Component", () => {
  let wrapper;

  it("should to render correctly", () => {
    const testState = {
      userReducer: {
        user: { firstAccess: true }
      }
    };

    const store = createMockStore(testState);
    wrapper = shallowWithStore(<LoginContainer />, store);
    expect(wrapper.dive().props().state).toEqual(testState.userReducer);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
