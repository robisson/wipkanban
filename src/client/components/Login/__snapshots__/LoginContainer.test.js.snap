// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginContainer /> Component should to render correctly 1`] = `
<WithStyles(Login)
  onLogin={[Function]}
  state={
    Object {
      "user": Object {
        "firstAccess": true,
      },
    }
  }
/>
`;
